src/components/ViewerNavigation/ViewerNavigation.tsx

```diff
@@ -93,6 +109,24 @@ const ViewerNavigation: FC<Props> = ({
         onBack={onBack}
       />
       <Aside>
+        {hasInterimQueries && (
+          <ClearFilters
+            onClick={() => {
+              dispatch(resetDatePickerWidgets(activeDashboard));
+              dispatch(resetFilterWidgets(activeDashboard));
```

Comment:
As this is 4x dispatch specific action - I suggest to move it as separate saga action - responsible for clearing filters. What do you think about it ?